### PR TITLE
Invites: add ability to resend pending invites

### PR DESCRIPTION
**Pull request by michaeldcain**
This adds the action, reducer, and selector to manage resending pending invites. Will do the UI implementation in a new PR.

Ref: p3Ex-32O-p2

### CODE REVIEW

File: client/state/invites/selectors.js [hunk 1]
```diff
@@ -27,3 +27,20 @@ export function getInvitesForSite( state, siteId ) {
...
+ * @param  {Number}  siteId Site ID
+ * @param  {String}  inviteId Invite ID
+ * @return {Boolean}        Whether invites are being requested
+ */
+export function isRequestingInviteResend( state, siteId, inviteId ) {
+	const siteResendRequests = state.invites.requestingInviteResend[ siteId ];
+	if ( ! siteResendRequests ) {
+		return null;
+	}
+	return !! siteResendRequests[ inviteId ];
```
Comment (sirbrillig) [inline client/state/invites/selectors.js]: I think you could also use `return get( state, [ 'invites', 'requestingInviteResend', siteId, inviteId ], false );`

File: client/state/invites/selectors.js [hunk 2]
```diff
@@ -27,3 +27,20 @@ export function getInvitesForSite( state, siteId ) {
...
+ *
+ * @param  {Object}  state  Global state tree
+ * @param  {Number}  siteId Site ID
+ * @param  {String}  inviteId Invite ID
+ * @return {Boolean}        Whether invites are being requested
+ */
+export function isRequestingInviteResend( state, siteId, inviteId ) {
+	const siteResendRequests = state.invites.requestingInviteResend[ siteId ];
+	if ( ! siteResendRequests ) {
+		return null;
```
Comment (sirbrillig) [inline client/state/invites/selectors.js]: Probably fine, but remember that `null` is not a Boolean. So this function as written has three possible return values: `true`, `false`, or `null`.

File: client/state/invites/selectors.js [hunk 3]
```diff
@@ -27,3 +32,16 @@ export function getInvitesForSite( state, siteId ) {
 	}
 	return invites;
 }
+
+/**
+ * Returns true if currently requesting an invite resend for the given site, or false
```
Comment (nylen) [inline client/state/invites/selectors.js]: "for the given site and invite ID"

File: client/state/invites/reducer.js [hunk 4]
```diff
@@ -52,4 +55,29 @@ export const items = createReducer(
...
+/**
+ * Returns the updated site invites resend requests state after an action has been
+ * dispatched. The state reflects an object keyed by site ID, consisting of requested
+ * resend invite IDs, with a boolean representing request status.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function requestingInviteResend( state = {}, action ) {
```
Comment (nylen) [inline client/state/invites/reducer.js]: I slightly prefer `state.invites.requestingResend` over the current `state.invites.requestingInviteResend`.

File: client/state/invites/test/reducer.js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+import deepFreeze from 'deep-freeze';
+
+/**
+ * Internal dependencies
+ */
+import { requestingResend } from '../reducer';
+import { INVITE_RESEND_REQUEST } from 'state/action-types';
+
+describe( 'reducer', () => {
+	describe( '#resendRequests()', () => {
```
Comment (nylen) [inline client/state/invites/test/reducer.js]: This should match the reducer name: `resendRequests`

File: client/state/invites/test/reducer.js [hunk 6]
```diff
@@ -0,0 +1,54 @@
...
+				{
+					type: INVITE_RESEND_REQUEST,
+					siteId: 12345,
+					inviteId: '123456asdf789',
+				}
+			);
+			expect( state ).to.eql( {
+				12345: { '123456asdf789': true },
+			} );
+		} );
```
Comment (nylen) [inline client/state/invites/test/reducer.js]: Super minor: I would add blank lines in between each test, for clear visual separation

File: client/state/invites/test/selectors.js [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+			};
+			expect( isRequestingResend( state, 12345, '123456asdf789' ) ).to.equal( false );
+		} );
+		test( 'should return false when resend has not been requested', () => {
+			const state = {
+				invites: {
+					requestingResend: {
+						12345: { '123456asdf789': false },
+						67890: { '789lkjh123456': true },
+					},
```
Comment (nylen) [inline client/state/invites/test/selectors.js]: This should be `requestingResend: {}`, no?

Comment (nylen) [reply client/state/invites/test/selectors.js]: When I left this comment, I missed that the invite ID tested here is different, but I made this change in https://github.com/Automattic/wp-calypso/pull/21845 to improve the clarity of this test.

File: client/state/invites/test/reducer.js (hunk 6 above)
Comment (nylen) [reply client/state/invites/test/reducer.js]: Done in https://github.com/Automattic/wp-calypso/pull/21845.

**Pull request merged**